File: static/mypusher.js
// Enable pusher logging - don't include this in production
//Pusher.logToConsole = true;

function getconfig()
{
    fetch('/getconfig', {
    method: 'GET',
    headers: {
        'Content-Type': 'application/json'
    },
    }).then(response => {
        if (response.ok) {
            // If the response status is in the 200-299 range, it means the request was successful.
            return response.json(); // Parse the response as JSON
        } else {
            // Handle errors or non-successful responses here
            throw new Error('Request failed with status: ' + response.status);
        }
    }).then(data => {

        console.log(data);

        // TODO: get pusher key from my server
        var pusher = new Pusher(data["PUSHER_API_KEY"], {
            cluster: 'us2'
        });
    
        var channel = pusher.subscribe('lineupgen-shareduser');
        channel.bind('game-changed', function(data) {
            // if this game_id is the one I have open then update it
            var current_game_id = document.getElementById('current_game_id');
            //alert(data.game_id + current_game_id.value);
            if (current_game_id.value == data.game_id)
            {
            console.log("other person edited - updating")
            getserverdata(data.game_id);
            }
    
        });


    }).catch(error => {
        console.error(error);
    });}

File: mypusher.py
import pusher
import json
from logger import log as log

json_file_path = 'mypusher.json'
with open(json_file_path, 'r') as file:
    pusher_info = json.load(file)


log.debug("pusher key " + pusher_info["key"])

pusher_client = pusher.Pusher(
  app_id=pusher_info["app_id"],
  key=pusher_info["key"],
  secret=pusher_info["secret"],
  cluster='us2',
  ssl=True
)

def notify_shared_users_of_game_change(game_id):
    pusher_client.trigger('lineupgen-shareduser', 'game-changed', {'game_id': game_id})